src/ScheduleFunctions.cpp

```diff
@@ -1075,13 +1073,13 @@ void validate_schedule(Function f, Stmt s, const Target &target, bool is_output)
 
     // Emit a warning if only some of the steps have been scheduled.
     bool any_scheduled = f.schedule().touched();
-    for (const UpdateDefinition &r : f.updates()) {
-        any_scheduled = any_scheduled || r.schedule.touched();
+    for (const Definition &r : f.updates()) {
+        any_scheduled = any_scheduled || r.schedule().touched();
     }
     if (any_scheduled) {
         for (size_t i = 0; i < f.updates().size(); i++) {
-            const UpdateDefinition &r = f.updates()[i];
-            if (!r.schedule.touched()) {
+            const Definition &r = f.update(i);
+            if (!r.schedule().touched()) {
                 std::cerr << "Warning: Update step " << i
```

Comment:
I know this is not changed code, but I just spotted this, it seems like maybe this should be user_warning << ...